x-pack/plugins/apm/public/components/app/transaction_details/trace_samples_tab.tsx

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { EuiSpacer } from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import { useUrlParams } from '../../../context/url_params_context/use_url_params';
+import { FETCH_STATUS } from '../../../hooks/use_fetcher';
+
+import { MlLatencyCorrelations } from '../correlations/ml_latency_correlations';
+
+import { useWaterfallFetcher } from './use_waterfall_fetcher';
+import type { TabContentProps } from './types';
+import { WaterfallWithSummary } from './waterfall_with_summary';
+
+function TraceSamplesTab({
+  selectSampleFromChartSelection,
+  clearChartSelection,
+  sampleRangeFrom,
+  sampleRangeTo,
+  traceSamples,
+}: TabContentProps) {
+  const { urlParams } = useUrlParams();
+
+  const {
+    waterfall,
+    exceedsMax,
+    status: waterfallStatus,
+  } = useWaterfallFetcher();
+
+  return (
+    <>
+      <MlLatencyCorrelations
+        correlationAnalysisEnabled={false}
+        onChartSelection={selectSampleFromChartSelection}
+        onClearSelection={clearChartSelection}
+        selection={
+          sampleRangeFrom !== undefined && sampleRangeTo !== undefined
+            ? [sampleRangeFrom, sampleRangeTo]
+            : undefined
+        }
+        markerCurrentTransaction={
+          waterfall.entryWaterfallTransaction?.doc.transaction.duration.us
+        }
+      />
```

Comment:
This only renders the distribution, and not the correlation analysis, right? If that's the case, can you extract the distribution from `MlLatencyCorrelations` to something like `TransactionDistribution` ?

Reply:
Yep, refactored in 539102e93f3. It resulted in a bit of code duplication but each component by itself should now be more simple. Note as part of that refactor I deleted the no longer needed pre-7.14 `latency_correlations.tsx` that was still around and renamed `ml_latency_correlations.tsx` to `latency_correlation.tsx` which resulted in a bit of an unfortunate diff for review :(.